content/what-is-roles-based-access-control-vs-abac/index.md

```diff
@@ -0,0 +1,66 @@
+---
+title: What is Roles-Based Access Control (RBAC)?
+date: "2022-10-20"
+description: "Learn about RBAC and advantages + disadvantages compared to ABAC."
+cover: "what-is-rbac-cover.png"
+category: "programming"
+author: "SuperTokens Team"
+---
+
+
+Roles-based access control (RBAC) is a security approach that uses roles to define what a user is and isn’t allowed to do. In a RBAC system, users are assigned roles with varying permissions for different resources, including files, databases, and applications.
+
+So, when a user tries to access a resource, the system will first find the roles associated with the user and then check if any of the roles have the appropriate permission. If so, the user is allowed to access the resource. If not, the user is denied access.
+
+For our example, we’ll be building a sample blogging app.
+
+First, we have a `regular-user` role. We’ll want to allow regular users to read all blog posts, but only edit or delete posts created by them.
+
+On the other hand, we create an `admin` role that allows admins to create, edit, and delete all blog posts.
+
+So if a user with only the `regular-user` role tried to edit a blog post by another user, our system would see that the `regular-user` role doesn’t have the permission to edit that blog post and deny the request. 
+
+## RBAC vs. ABAC?
+
+As opposed to RBAC, Attribute-based access control (ABAC) is another security approach that assigns permissions based on user attributes, resource attributes, and environment attributes. For example, a design file can be restricted to users that have designer in their title and only accessed on weekdays.
+
+While ABAC gives finer granularity over access to different resources, RBAC wins out on the ease of implementation. The initial setup cost of RBAC is magnitudes lower than ABAC.
+
+With RBAC, companies have a straightforward method for grouping users with similar access needs and then assigning permissions to those groups.
+
+With ABAC, companies must first define variables and configure different attribute-based rules, which can be a cumbersome process.
+
+![RBAC vs ABAC](./rbac-abac.png)
+
+
+## Advantages of RBAC
+
+Easy to understand: Because of its intuitive structure, the underlying business logic with RBAC is simple to communicate and understand – even with dozens of different roles.
```

Comment:
```suggestion
- Easy to understand: Because of its intuitive structure, the underlying business logic with RBAC is simple to communicate and understand – even with dozens of different roles.
```